FusionIIIT/applications/complaint_system/views.py

```diff
@@ -26,64 +28,49 @@ def complaint_reassign(request,wid,iid):
+            w = Workers.objects.select_related('secincharge_id','secincharge_id__staff_id','secincharge_id__staff_id__user','secincharge_id__staff_id__department').get(id=worker_id)
             StudentComplain.objects.select_for_update().filter(id=iid).\
                 update(worker_id=w, status=1)
-            url = '/complaint/caretaker/worker_id_know_more/'+wid;
-            complainer_details = StudentComplain.objects.select_related('complainer','complainer__user','complainer__department','worker_id','worker_id__caretaker_id__staff_id','worker_id__caretaker_id__staff_id__user','worker_id__caretaker_id__staff_id__department').get(id=iid)
+            url = '/complaint/secincharge/worker_id_know_more/'+wid;
+            complainer_details = StudentComplain.objects.select_related('complainer','complainer__user','complainer__department','worker_id','worker_id__secincharge_id__staff_id','worker_id__secincharge_id__staff_id__user','worker_id__secincharge_id__staff_id__department').get(id=iid)
             student=0
             message = "Your complaint has been re-assigned"
             complaint_system_notif(request.user, complainer_details.complainer.user ,'reassign_worker_alert',complainer_details.id,student,message)
             return HttpResponseRedirect(url)
-        elif type == 'redirect':
-            assign_caretaker = request.POST.get('assign_caretaker', '')
-            c = Caretaker.objects.select_related('staff_id','staff_id__user','staff_id__department').get(id=assign_caretaker)
-            c1 = Caretaker.objects.select_related('staff_id','staff_id__user','staff_id__department').get(id=iid)
-            remark = 'Redirect complaint from ' + c1.area
-            StudentComplain.objects.select_for_update().filter(id=iid).\
-                update(location=c.area, remarks=remark)
-            url = '/complaint/caretaker/worker_id_know_more/'+wid;
-            complainer_details = StudentComplain.objects.select_related('complainer','complainer__user','complainer__department','worker_id','worker_id__caretaker_id__staff_id','worker_id__caretaker_id__staff_id__user','worker_id__caretaker_id__staff_id__department').get(id=iid)
-            student=0
-            message = "Your complaint has been redirected to another caretaker"
-            complaint_system_notif(request.user, complainer_details.complainer.user ,'comp_redirect_alert',complainer_details.id,student,message)
-            return HttpResponseRedirect(url)
         
     else:
         y = ExtraInfo.objects.all().select_related('user','department').get(id=y.id)
-        a = Caretaker.objects.select_related('staff_id','staff_id__user','staff_id__department').get(staff_id=y)
-        b = a.area
+        a = SectionIncharge.objects.select_related('staff_id','staff_id__user','staff_id__department').get(staff_id=y)
+        b = a.work_type

```

Comment:
You can use better variable names instead of letters which will improve code readability.